Strings/string_permutations.cpp

```diff
@@ -0,0 +1,50 @@
+// Problem: Print all permutations of string
+
+// Input: ABC
+// Output: ABC ACB BAC BCA CBA CAB
+
+// Concept: Backtracking
+
+#include<bits/stdc++.h>
+#define fl(i,a,b) for(i=a;i<b;i++)
+#define fast ios_base::sync_with_stdio(false);cin.tie(NULL);cout.tie(NULL);
+using namespace std;
+typedef long long int ll;
+vector<string> permutations;
+
+vector<string> permute(string str, ll l, ll r)
+{
+   if(l==r)
+      {
+         permutations.push_back(str);
+         return permutations;
+      }
+   for(ll i=l; i<=r; i++)
+   {
+      swap(str[l], str[i]);
+      permute(str, l+1, r);
+      swap(str[l], str[i]);
+   }
+   return permutations;
+}
+
+int main()
+{ 
+   fast;
+
+// #ifndef ONLINE_JUDGE
```

Comment:
Great work. Just delete the unwanted commented lines that you probably used for debugging purposes.

Reply:
Done